subsys/net/lib/coap/coap_transport_socket.c

```diff
@@ -176,6 +176,32 @@ static session_t *session_find(const struct sockaddr *local,
 }
 #endif /* (COAP_SESSION_COUNT > 0) */
 
+
+/**@brief Internal method to find the session or port corresponding to the handle.
+ *
+ * @param handle Identifies the socket descriptor, also the the handle for the transport.
+ *
+ * @retval Pointer to a valid transport in case of success, else, NULL.
+ */
+static int32_t local_endpoint_find(coap_transport_handle_t  handle)
+{
+    // Search only the ports, not the sessions.
+    for (int index = 0; index < COAP_SOCKET_COUNT; index++) {
+        if (port_table[index].socket_fd == handle) {
+            return index;
+        }
+    }
+
+    return -1;
+}
+
+
+static bool secure_endpoint_check(uint32_t index)
+{
+    index > COAP_PORT_COUNT ? true : false;
```

Comment:
Shouldn't it be `>=`? I guess non-secure sockets will have indexes from `0` to `COAP_PORT_COUNT - 1` ?